src/utils/Utils.ts

```diff
@@ -1607,4 +1608,92 @@ export default class Utils {
   public static hash(data: string): string {
     return crypto.createHash('sha256').update(data).digest('hex');
   }
+
+  public static findSiteAreaInTrees(siteAreaID: string, siteAreas: SiteArea[] = null, siteAreaTreeNode: SiteArea = null): SiteArea {
+    if (siteAreaTreeNode) {
+      // Check if current site area is the one to find
+      if (siteAreaID === siteAreaTreeNode.id) {
+        return siteAreaTreeNode;
+      }
+      if (!Utils.isEmptyArray(siteAreaTreeNode.siteAreaChildren)) {
+        // Check all children and children of children
+        for (const siteAreaChild of siteAreaTreeNode.siteAreaChildren) {
+          if (siteAreaChild.id === siteAreaID) {
+            return siteAreaTreeNode;
+          }
+          if (this.findSiteAreaInTrees(siteAreaID, null, siteAreaChild)) {
+            return siteAreaTreeNode;
+          }
+        }
+      }
+      return null;
+    }
+    if (Array.isArray(siteAreas)) {
+      // Loop through trees to find specific tree
+      for (const siteArea of siteAreas) {
+        // If site area ID is defined return tree, which contains the site area
+        const requestedTree = this.findSiteAreaInTrees(siteAreaID, null, siteArea);
+        if (requestedTree) {
+          return siteArea;
+        }
+      }
+    }
+  }
+
+  public static buildSiteAreaTrees(siteAreas: SiteArea[]): SiteArea[] {
+    // Hash Table helper
+    const hashTable = {};
+    for (const siteArea of siteAreas) {
+      hashTable[siteArea.id] = { ...siteArea, siteAreaChildren: [] } as SiteArea;
+    }
+    const siteAreaTrees: SiteArea[] = [];
+    // Build tree
+    for (const siteArea of siteAreas) {
+      if (!Utils.isNullOrUndefined(siteArea.parentSiteAreaID) && !Utils.isNullOrUndefined(hashTable[siteArea.parentSiteAreaID])) {
+        // Push sub site area to parent children array
+        hashTable[siteArea.parentSiteAreaID].siteAreaChildren.push(hashTable[siteArea.id]);
+      } else {
+        // If no parent ID is defined push root site area to array
+        siteAreaTrees.push(hashTable[siteArea.id]);
+      }
+    }
+    return siteAreaTrees;
+  }
+
+  public static checkSiteAreaTrees(siteAreaTrees: SiteArea[], numberOfSiteAreas?: number,
+      siteAreaTree: SiteArea = null, count = 0) : number {
+    if (siteAreaTree) {
+      count++;
+      // Count and check all children and children of children
+      if (!Utils.isEmptyArray(siteAreaTree.siteAreaChildren)) {
+        for (const siteAreaChild of siteAreaTree.siteAreaChildren) {
+          if (siteAreaTree.smartCharging !== siteAreaChild.smartCharging ||
+            siteAreaTree.numberOfPhases !== siteAreaChild.numberOfPhases ||
+            siteAreaTree.siteID !== siteAreaChild.siteID ||
+            siteAreaTree.voltage !== siteAreaChild.voltage) {
+            throw new BackendError({
+              siteAreaID: siteAreaChild.id,
+              method: 'checkSiteAreaTrees',
+              message: 'Property inconsistency in Site Area Tree',
```

Comment:
Add detailedMessage to display what was wrong like:
`Expected numberOfPhases '3' from Site Area 'sss' but got '2' in Site Area 'ddd'`
Help yourself in prod :)